libvncserver/tight.c

```diff
@@ -215,12 +215,12 @@ static int PaletteInsert (uint32_t rgb, int numPixels, int bpp);
 static void Pack24 (rfbClientPtr cl, char *buf, rfbPixelFormat *fmt,
                     int count);
 
-static void EncodeIndexedRect16 (uint8_t *buf, int count);
-static void EncodeIndexedRect32 (uint8_t *buf, int count);
+static void EncodeIndexedRect16 (uint8_t *buf, int count, uint16_t mask);
```

Comment:
If that fix is legit, the bug goes back to 757fdc2454a97890c119cca8662b8d1763265e9a which used the original GPL dump. So other tight encoders might be affected as well? Or have already fixed this?

Reply:
Just checked, no mask at least in https://www.tightvnc.com/download/1.3.10/tightvnc-1.3.10_unixsrc.tar.bz2

Reply:
Maybe it's totally invalid for the X11 server to have "garbage" outside the RGB mask in the framebuffer?

Reply:
Mhh, that I don't know. Maybe you can try to repro with Turbo- or TightVNC in your setup and, in case the bug is reproducible, report the issue there as well? Could be that you spotted something that affects a lot of projects...